Improve user-friendliness in mortar gap heat transfer action

recuero opened the pull request:
Better parameter information in mortar gap heat transfer action/objects (#20922)

Since the gap heat transfer mortar action builds physics objects in various ways (thru action or user objects), users can be confused as to which parameters really count internally. This PR improves user interface and avoids some misleading combinations.

recuero committed:
Better parameter information in mortar gap heat transfer action/objects (#20922)

recuero committed:
Add gap conductance function test with mortar gap heat transfer action

The objects allow users to provide a constant gap conductance and then a
function that multipliers such constant gap conductance. This doesn't seem
intuitive to me, but it's been there for a while so I am leaving it as is.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1053124/) on 2e6fcaf wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/21103/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1053133/) on 2e6fcaf wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>c325aa</th>
      <th colspan=4>#21103 2e6fca</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.58%</td>
      <td>82.57%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>75282</td>
      <td>75281</td>
      <td>-1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15886</td>
      <td>15887</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/21103/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/21103/coverage/framework)


## Modules coverage

### Heat conduction

<table>
  <thead>
    <tr>
      <th></th>
      <th>c325aa</th>
      <th colspan=4>#21103 2e6fca</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.98%</td>
      <td>90.00%</td>
      <td>+0.03%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>3196</td>
      <td>3205</td>
      <td>+9</td>
      <td>9</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>356</td>
      <td>356</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/21103/coverage_diff/heat_conduction)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/21103/coverage/heat_conduction)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/21103/coverage/xfem)

</details>



This comment will be updated on new commits.

recuero committed:
Make gap conductance value default to one.

This change makes it consistent with the legacy point on face approach to
solving the gap heat transfer problem.

Users can provide a constant gap conductance value and a function (possibly of temperature) which
 will be multiplied to come up with the value used.
